sphinx/source/docs/user_guide/interaction/callbacks.rst

```diff
@@ -77,10 +77,57 @@ is executed to update some data:
     :source-position: above
 
 
+
+CustomJS for Event classes
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In addition to responding to Backbone events using ``js_on_change``,
+Bokeh allows ``CustomJS`` callbacks to be triggered by a set of
+dedicated event objects. These events include interaction events with
+the plot canvas, button click events and LOD events.
+
+When registering with these events, the ``CustomJS`` callbacks is
+attached to the event using the ``js_on_event`` method of Bokeh models
+and the callback receives the event object as ``cb_obj``:
+
+
+.. code:: python
+
+    from bokeh.models.callbacks import CustomJS
+
+    callback = CustomJS(code="""
+    // the event that triggered the callback is cb_obj:
+    var event = cb_obj.value;
+    // The event type determines the relevant attributes
+    console.log('Tap event occured at x-position: ' + event.x)
+    """)
+
+    p = figure()
+    # execute a callback whenever the plot canvas is tapped
+    p.js_on_event('tap', callback)
+
+
+Although the event can be specified as a string such as ``'tap'`` above,
+it is often more appropriate to specify the first argument of
```

Comment:
I'm looking for a way to convey that the first arguments can be strings, but that passing the event class is a more robust way of specifying things (which is why I think I would recommend this style).

We should probably decide whether we are happy seeing strings in ``js_on_event`` and ``on_event`` or whether we might prefer to always specify via event class. The only real advantage of using strings that I see is it helps avoid one import...